pkg/tsdb/azuremonitor/aztokenprovider/token_provider.go

```diff
@@ -132,7 +132,11 @@ func (c *managedIdentityTokenRetriever) GetCacheKey() string {
 }
 
 func (c *managedIdentityTokenRetriever) Init() error {
-	if credential, err := azidentity.NewManagedIdentityCredential(c.clientId, nil); err != nil {
+	opts := &azidentity.ManagedIdentityCredentialOptions{
+		ID: azidentity.ClientID(c.clientId),
```

Comment:
I see there are two different kinds of ids now: `azidentity.ClientID` and `azidentity.ResourceID`. 

Based on the field name (`clientId`), I decided to go with `azidentity.ClientID`, but I'm wondering if that's same or if we should add some mechanism to determine if it's one or the other.

cc/ @sunker

Reply:
yes, I believe we only support a clientID here, also, from the comment: 

```
The value may be the identity's client ID or resource ID, but note that some platforms don't accept resource IDs.
```

Reply:
I believe prometheus uses `resourceId` for template variable queries. But that is just temporary right @kostrse? Should be fine to use client id in all scenarios?

Reply:
In this context the `ClientID` and `ResourceID` are both the ways of identification of the managed identity.

In our case only `ClientID` is exposed to the user:
* https://github.com/grafana/grafana/blob/6889e39526e1b71254c9030ea4ac6d821a21cb17/pkg/setting/setting_azure.go#L15
* https://github.com/grafana/grafana/blob/6889e39526e1b71254c9030ea4ac6d821a21cb17/conf/defaults.ini#L631

You can ignore `ResourceID`

Reply:
[How to use managed identities for Azure resources on an Azure VM to acquire an access token](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http)
`client_id` and `mi_res_id`